ledger-core/v2/virtual/handlers/vdelegatedrequestfinished.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package handlers
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/injector"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/descriptor"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/object"
+)
+
+type SMVDelegatedRequestFinished struct {
+	// input arguments
+	Meta    *payload.Meta
+	Payload *payload.VDelegatedRequestFinished
+
+	// dependencies
+	objectCatalog object.Catalog
+}
+
+var dSMVDelegatedRequestFinishedInstance smachine.StateMachineDeclaration = &dSMVDelegatedRequestFinished{}
+
+type dSMVDelegatedRequestFinished struct {
+	smachine.StateMachineDeclTemplate
+}
+
+func (*dSMVDelegatedRequestFinished) InjectDependencies(sm smachine.StateMachine, _ smachine.SlotLink, injector *injector.DependencyInjector) {
+	s := sm.(*SMVDelegatedRequestFinished)
+
+	injector.MustInject(&s.objectCatalog)
+}
+
+func (*dSMVDelegatedRequestFinished) GetInitStateFor(sm smachine.StateMachine) smachine.InitFunc {
+	s := sm.(*SMVDelegatedRequestFinished)
+	return s.Init
+}
+
+/* -------- Instance ------------- */
+
+func (s *SMVDelegatedRequestFinished) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return dSMVDelegatedRequestFinishedInstance
+}
+
+func (s *SMVDelegatedRequestFinished) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	return ctx.Jump(s.stepProcess)
+}
+
+func (s *SMVDelegatedRequestFinished) stepProcess(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	objectRef := s.Payload.Callee
+
+	var objectDescriptor *descriptor.Object
+
+	if s.Payload.LatestState != nil {
+		state := s.Payload.LatestState
+		desc := descriptor.NewObject(
+			objectRef,
+			state.Reference,
+			state.Prototype,
+			state.State,
+			state.Parent,
+		)
+		objectDescriptor = &desc
+	}
+
+	sharedObjectState := s.objectCatalog.Get(ctx, objectRef)
```

Comment:
Get raises 'panic' if there is no such object.
What if DelegatedRequestFinished comes before any other requests/messages for this object?

Reply:
@ivanshibitov As far I know we can not get VDelegatedRequestFinished before VStateReport.

Reply:
Ideally, we have to be ready to such things. 
But only @cyraxred knows the truth

Reply:
@ivanshibitov is absolutely right, SM of this message even should wait for object to be ready like execute does. We can not rely on order of two messages. VPendingFinished can come faster than VStateReport.

Reply:
@ruz Fixed